sled-agent/src/illumos/vnic.rs

```diff
@@ -86,6 +86,7 @@ impl<DL: VnicSource + Clone> VnicAllocator<DL> {
 /// communicating with Oxide services.
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum VnicKind {
+    Physical,
```

Comment:
The logic in the zone creation is identical for vnics and physical nics, so I didn't want to complicate the interface for no functional reason.  In fact, the zone `install()` calls the argument `physical_nic: Option<Vnic>`.  I'm going to rename Vnic -> Link, which will resolve the existing ambiguity and prevent me from making it worse.

Reply:
I agree that change is beyond the scope of this PR. The addition of a `Physical` variant to the `Vnic` enum just made clear that it's not really the right type or abstraction here. Nothing for you to do, but it might make sense to create an issue to clean this up.